### PR TITLE
Try/monticello block menu patterns

### PR DESCRIPTION
**Pull request by pbking**
A first pass a creating re-usable elements that can be used to build menus for Monticello block-based theme.

In block-patterns.php I registered a NEW pattern called 'Menu Item'.  
This was mostly gutenberg markup I coped out of a page editor and tweaked by hand a little bit as I worked through it.  It's essentially a GROUP at the top-level wrapping a COLUMNS container with three pieces of content (name, description, price).  The top-level GROUP has the `is-style-menu-list-item` class applied out-of-the-gate so that the block style (outlined later) is picked up as soon as it's dropped in.  The other P elements all have helpful custom classes assigned for further styling.

In block-styles.php (added and called from functions.php) the custom block styles are initialized and pulled in from block-styles.css and the `menu-list-item` block style is defined.

block-styles.css provides the custom CSS that the `menu-list-item` needs to lay out properly in this theme.

The above registers everything and allows me to drop reusable elements into my page that I can pretty easily change to suit the needs outlined in the comp.  The result, after some work, looks something like this:

image

image

I noticed a couple of limitations while leveraging these elements to build out a page:

* While it's simple to add these elements onto a PAGE it's not possible to easily add them into a CONTAINER.  When attempting to add menu items to my two-column section I was NOT offered the options of choosing one of the patterns.
image

* Using the Gutenberg editor to edit a POST I was able to leverage this Block Pattern.  However, using the Full Site Editor I had no option to choose to insert a Block Pattern.
image

### CODE REVIEW

**Comment by pbking** [comment]
I also had a question as to the appropriateness of adding this new "group style" to a theme.
image

These tools gave me a pretty good method of styling a section of elements, and a pretty easy way (mostly) of adding that section of elements to a page such that they will be styled as expected.  But in doing so it feels like we're "muddying the waters" for "group" themes.  Meaning: This thing we're adding, it isn't something that you could/would apply to ANY 'group' to change the style to suit.  It's more about changing things SPECIFICALLY for this use-case and I'm not sure that using a 'block-style' in this instance is actually helpful.  Simply including "custom CSS for this stamp-able block of content in styles.css" seems more appropriate to me.

File: monticello/style.css [hunk 1]
```diff
@@ -81,4 +81,19 @@ represented in .json
 	opacity: 1;
 }
 
-/* END BUTTON STYLES */
\ No newline at end of file
+/* END BUTTON STYLES */
+/*
```
Comment (jffng) [inline monticello/style.css:86]: Can we remove these commented styles?

**Review by jffng — APPROVED** [review]
> While it's simple to add these elements onto a PAGE it's not possible to easily add them into a CONTAINER. When attempting to add menu items to my two-column section I was NOT offered the options of choosing one of the patterns.

It's true, the pattern insert does not appear when it's in a container. You can search for it however: 

Screen Shot 2020-11-18 at 8 40 59 PM

> Using the Gutenberg editor to edit a POST I was able to leverage this Block Pattern. However, using the Full Site Editor I had no option to choose to insert a Block Pattern.

Certain blocks are allowed / disallowed depending on the context (full site editor vs page/post). Do you think we should be able to able to insert block patterns in the site editor? 

> It's more about changing things SPECIFICALLY for this use-case and I'm not sure that using a 'block-style' in this instance is actually helpful. 

I think I agree with you, it seems like an abuse of the block styles system to apply such specific menu styles to a group when it's such a generic block.

File: monticello/style.css (hunk 1 above)
Comment (pbking) [reply monticello/style.css:86]: I think (based on the conversation above) I'm about to UNcomment them and pull out the block styles.

**Pull request merged**